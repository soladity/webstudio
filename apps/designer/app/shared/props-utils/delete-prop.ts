import { type AllUserProps, type DeleteProp } from "@webstudio-is/react-sdk";

export const deletePropMutable = (
  allProps: AllUserProps,
  deleteProp: DeleteProp
) => {
  const prop = allProps[deleteProp.instanceId];
  if (prop === undefined) {
    return false;
  }
  const index = prop.props.findIndex(({ id }) => id === deleteProp.propId);
  if (index === -1) {
    return false;
  }
  prop.props.splice(index, 1);
  return true;
};
