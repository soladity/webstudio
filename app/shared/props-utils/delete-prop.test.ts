import { type AllUserProps } from "@webstudio-is/sdk";
import { deletePropMutable } from "./delete-prop";

describe("Delete prop", () => {
  test("found and deleted", () => {
    const propsMap: AllUserProps = {
      instanceId: {
        id: "id",
        instanceId: "instanceId",
        treeId: "treeId",
        props: [
          { id: "propId", prop: "b", value: "1" },
          { id: "propId1", prop: "a", value: "1" },
        ],
      },
    };

    const isDeleted = deletePropMutable(propsMap, {
      instanceId: "instanceId",
      propId: "propId",
    });
    expect(propsMap).toMatchSnapshot();
    expect(isDeleted).toBeTruthy();
  });

  test("not found and not deleted", () => {
    const propsMap: AllUserProps = {
      instanceId: {
        id: "id",
        instanceId: "instanceId",
        treeId: "treeId",
        props: [
          { id: "propId", prop: "b", value: "1" },
          { id: "propId1", prop: "a", value: "1" },
        ],
      },
    };

    const isDeleted = deletePropMutable(propsMap, {
      instanceId: "instanceId",
      propId: "notFound",
    });
    expect(propsMap).toMatchSnapshot();
    expect(isDeleted).toBeFalsy();
  });
});
