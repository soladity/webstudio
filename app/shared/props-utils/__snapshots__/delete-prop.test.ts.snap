// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete prop found and deleted 1`] = `
Object {
  "instanceId": Object {
    "id": "id",
    "instanceId": "instanceId",
    "props": Array [
      Object {
        "id": "propId1",
        "prop": "a",
        "value": "1",
      },
    ],
    "treeId": "treeId",
  },
}
`;

exports[`Delete prop not found and not deleted 1`] = `
Object {
  "instanceId": Object {
    "id": "id",
    "instanceId": "instanceId",
    "props": Array [
      Object {
        "id": "propId",
        "prop": "b",
        "value": "1",
      },
      Object {
        "id": "propId1",
        "prop": "a",
        "value": "1",
      },
    ],
    "treeId": "treeId",
  },
}
`;
